src/components/EmojiPicker/EmojiPickerMenu/getItemType.js

```diff
@@ -0,0 +1,22 @@
+import CONST from '@src/CONST';
+
+/**
+ * Improves FlashList's recycling when there are different types of items
+ * @param {Object} item
+ * @returns {String}
+ */
+function getItemType(item) {
+    // item is undefined only when list is empty
+    if (!item) {
+        return;
+    }
+    if (item.header) {
+        return CONST.EMOJI_PICKER_ITEM_TYPES.HEADER;
+    }
+    if (item.spacer) {
+        return CONST.EMOJI_PICKER_ITEM_TYPES.SPACER;
+    }
+    return CONST.EMOJI_PICKER_ITEM_TYPES.EMOJI;
```

Comment:
If you're really trying to hyper-optimize this we should check this (the most common case) before checking for headers and spacers